aws-redshiftserverless-namespace/src/main/java/software/amazon/redshiftserverless/namespace/ReadHandler.java

```diff
@@ -136,19 +153,31 @@ private GetResourcePolicyResponse getNamespaceResourcePolicy(
         return getResponse;
     }
 
+    private ProgressEvent<ResourceModel, CallbackContext> listSnapshotCopyConfigurationErrorHandler(final ListSnapshotCopyConfigurationsRequest request,
+                                                                                                    final Exception exception,
+                                                                                                    final ProxyClient<RedshiftServerlessClient> client,
+                                                                                                    final ResourceModel model,
+                                                                                                    final CallbackContext context) {
+        if (exception instanceof ValidationException) {
+            // ValidationException is thrown when the feature is not enabled in a region
+            logger.log(String.format("CRC feature is not enabled for this region: %s", exception.getMessage()));
+            return ProgressEvent.defaultSuccessHandler(model);
+        } else if (!containsSnapshotCopyConfigurations) {
```

Comment:
instead of calling the list API always then check if model contains CopyConfiguration, can we check the containsCopyConfiguration first then decide if we want to make the API call at all?

Reply:
my bad, we always need to fetch the current model first